parlai/tasks/coqa/agents.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import ParlAIDialogTeacher
+from .build import build
+
+import copy
+import os
+
+
+def _path(opt):
+    # Build the data if it doesn't exist.
+    build(opt)
+    dt = opt['datatype'].split(':')[0]
+    return os.path.join(opt['datapath'], 'CoQA', dt + '.txt')
+
+
+class DefaultTeacher(ParlAIDialogTeacher):
+    def __init__(self, opt, shared=None):
+        opt = copy.deepcopy(opt)
```

Comment:
note: there is a deepcopy in the "Teacher" class. do you need this here?

Reply:
I am not sure if I understand its usage correctly, but I see https://github.com/facebookresearch/ParlAI/blob/d80573384feb589f4246bba39e6c2ec85cefb97b/parlai/tasks/convai2_wild_evaluation/agents.py#L22 here it also uses a deepcopy?

Reply:
Leave the deepcopy for now I think.